tsconfig.json

```diff
@@ -10,7 +10,8 @@
 		"resolveJsonModule": true,
 		"skipLibCheck": true,
 		"strict": true,
-		"target": "ES2022"
+		"target": "ES2022",
+		"useDefineForClassFields": false
```

Comment:
I believe this is always true when target is `ES2022`. https://www.typescriptlang.org/tsconfig#useDefineForClassFields

that said, I believe it might not have much of a meaning, as fields are not transpiled with that target.

Reply:
Oh, you are in for a surprise: Just try it in the TS Playground. If you set the target to ES2022, then class fields are only rendered as class fields if you (explicitly) set `useDefineForClassFields` to `true`.
https://www.typescriptlang.org/play?useDefineForClassFields=true&target=9#code/MYGwhgzhAEBiD29oG8BQ0PVJCsCWApiACbQC80ALgE4CuBA3KgL5A

If you set it to `false`, they are instead assigned in the constructor.
https://www.typescriptlang.org/play?target=9&useDefineForClassFields=false#code/MYGwhgzhAEBiD29oG8BQ0PVJCsCWApiACbQC80ALgE4CuBA3KgL5A

I did not like the effect of `true` before ES2022 because it created huge overhead in the generated code, but I hoped that native class fields would be faster. But apparently they are not, at least not yet, and the performance gains are striking.

Reply:
Sorry for not including the detailed effect of this PR. Yes, the default of `useDefineForClassFields` is `true` if the target is `ES2022`, Howerver we need to explicitly set it to `false` to place all class fields within the constructor during TS compilation. in that case, we could reduce Rollup runtime.

The performance before
<img width="547" alt="before" src="https://github.com/rollup/rollup/assets/33021497/bcb45e53-2a8b-4889-9ac0-8107fa1d59b9">

The performance after changes
<img width="547" alt="now" src="https://github.com/rollup/rollup/assets/33021497/7fb3f04e-a596-48d5-a437-86384ae28ce2">

Reply:
> Oh, you are in for a surprise: Just try it in the TS Playground. If you set the target to ES2022, then class fields are only rendered as class fields if you (explicitly in the Playground) set `useDefineForClassFields` to `true`. [typescriptlang.org/play?useDefineForClassFields=true&target=9#code/MYGwhgzhAEBiD29oG8BQ0PVJCsCWApiACbQC80ALgE4CuBA3KgL5A](https://www.typescriptlang.org/play?useDefineForClassFields=true&target=9#code/MYGwhgzhAEBiD29oG8BQ0PVJCsCWApiACbQC80ALgE4CuBA3KgL5A)
> 
> If you set it to `false`, they are instead assigned in the constructor. [typescriptlang.org/play?target=9&useDefineForClassFields=false#code/MYGwhgzhAEBiD29oG8BQ0PVJCsCWApiACbQC80ALgE4CuBA3KgL5A](https://www.typescriptlang.org/play?target=9&useDefineForClassFields=false#code/MYGwhgzhAEBiD29oG8BQ0PVJCsCWApiACbQC80ALgE4CuBA3KgL5A)

I wonder if this is a bug in typescript, or intentional behavior? 🤔   I would expect typescript to not transpile, regardless of the `useDefineForClassFields` setting when he target is `ES2022/ESNext`.

> I did not like the effect of `true` before ES2022 because it created huge overhead in the generated code, but I hoped that native class fields would be faster. But apparently they are not, at least not yet, and the performance gains are striking.

this is surprising: https://v8.dev/blog/faster-class-features